aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -46,25 +46,35 @@
      *                       durably store offline changes until
      *                       then can be written to the network
      * @param appSync An AppSync Endpoint
+     * @param baseSyncIntervalProvider Provider the interval in which at most
+     *                                 one base sync may occur
      */
     public Orchestrator(
             @NonNull final ModelProvider modelProvider,
             @NonNull final ModelSchemaRegistry modelSchemaRegistry,
             @NonNull final LocalStorageAdapter localStorageAdapter,
-            @NonNull final AppSync appSync) {
+            @NonNull final AppSync appSync,
+            @NonNull final SyncProcessor.BaseSyncIntervalProvider baseSyncIntervalProvider) {
         Objects.requireNonNull(modelSchemaRegistry);
         Objects.requireNonNull(modelProvider);
         Objects.requireNonNull(appSync);
         Objects.requireNonNull(localStorageAdapter);
 
         RemoteModelMutations remoteModelMutations = new RemoteModelMutations(appSync, modelProvider);
-        RemoteModelState remoteModelState = new RemoteModelState(appSync, modelProvider);
```

Comment:
Thanks for getting rid of this - I'm a huge proponent of not overly separating things out since it leads to endless rabbit trails to find the code you're looking for so I fully support consolidating this functionality into the SyncProcessor.

Reply:
This one did take it too far. The Sync Processor and Remote Model State concepts had a lot of the same concerns. This became clear when I needed to lookup/save sync times. Basically, I was going to have to weave two Rx chains together. So I just flattened all the crap into the same Java document.